web/html/src/manager/maintenance/calendar/web-calendar.tsx

```diff
@@ -0,0 +1,262 @@
+import * as React from "react";
+import { useState, useEffect, useRef } from "react";
+
+import FullCalendar from "@fullcalendar/react";
+import dayGridPlugin from "@fullcalendar/daygrid";
+import timeGridPlugin from "@fullcalendar/timegrid";
+import interactionPlugin from "@fullcalendar/interaction";
+import allLocales from "@fullcalendar/core/locales-all";
+
+import { MessageType, Utils as MessagesUtils } from "components/messages";
+import Network from "utils/network";
+
+type WebCalendarProps = {
+  id: number;
+  type: string;
+  messages: (messages: MessageType[]) => void;
+  clearMessages: (messages: void) => void;
+  responseError: (messages: MessageType[]) => void;
+};
+
+type Event = {
+  start: string;
+  end: string;
+  title: string;
+}
+
+const WebCalendar = (props: WebCalendarProps) => {
+
+  const [events, setEvents] = useState<Event[]>([]);
+  const [currentDate, setCurrentDate] = useState<Date>();
+  const calendarRef = useRef<any>();
+
+  useEffect(() => {
+    getEvents("initial");
+    setCurrentDate(getApi().currentDataManager.data.currentDate);
+    // Hack: Make Fullcalendar button icons show correctly
+    jQuery(".fc-backButton-button").html('<div class="fa fa-angle-left"></div>');
+    jQuery(".fc-nextButton-button").html('<div class="fa fa-angle-right"></div>');
+    jQuery(".fc-skipBackButton-button").html('<div class="fa fa-angle-double-left"></div>')
+      .prop('title', "Skip to the last maintenance window");
```

Comment:
Here and below, needs `t()`.

Reply:
Uff I missed those :sweat_smile:
Thanks for pointing out